llvm/include/llvm/Object/DXContainer.h

```diff
@@ -117,14 +117,24 @@ template <typename T> struct ViewArray {
 
 namespace DirectX {
 
+struct RootParameter {
+  dxbc::RootParameterHeader Header;
+  union {
+    dxbc::RootConstants Constants;
+  };
+};
 class RootSignature {
 private:
-  uint32_t Version;
-  uint32_t NumParameters;
-  uint32_t RootParametersOffset;
-  uint32_t NumStaticSamplers;
-  uint32_t StaticSamplersOffset;
-  uint32_t Flags;
+  uint32_t Version = 2;
+  uint32_t NumParameters = 0;
+  uint32_t RootParametersOffset = 0;
+  uint32_t NumStaticSamplers = 0;
+  uint32_t StaticSamplersOffset = 0;
+  uint32_t Flags = 0;
+
+  SmallVector<DirectX::RootParameter> Parameters;
```

Comment:
The downside to this approach is that reading a DXContainer (even if we don't actually look at the root signature) ends up parsing out and generating a copy of all the root parameters.

That seems like a lot of overkill.

Could we instead use ViewArray to make this both lower on memory and lazy?